### PR TITLE
DEPR: Deprecate numpy argument in read_json

### LINKED ISSUE #28512: DEPR: Deprecate numpy argument in read_json
I've never really been clear on the purpose of the `numpy` argument in to_json. Some digging brought me here https://github.com/pandas-dev/pandas/pull/3876#issuecomment-19390383 where it is explained that this maintains some kind of sequence to elements. To illustrate the only difference I could find

```python
>>> pd.read_json('[{"a": 1, "b": 2}, {"b":2, "a" :1}]',numpy=False, orient='records')
   a  b
0  1  2
1  1  2

>>> pd.read_json('[{"a": 1, "b": 2}, {"b":2, "a" :1}]',numpy=True, orient='records')
   a  b
0  1  2
1  2  1
```

I might be missing the point but I don't understand why this would be useful. Objects in JSON are by definition not ordered, so this is non-compliant and I think just plain confusing.

So I think good to deprecate unless anyone has objections.

### PR DESCRIPTION
**Pull request by lucaionescu**
- [x] closes #28512 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -1767,3 +1767,8 @@ def test_json_indent_all_orients(self, orient, expected):
     def test_json_negative_indent_raises(self):
         with pytest.raises(ValueError, match="must be a nonnegative integer"):
             pd.DataFrame().to_json(indent=-1)
+
+    def test_deprecate_numpy_argument_read_json(self):
+        df = DataFrame([1, 2, 3])
```
Comment (gfyoung) [inline pandas/tests/io/json/test_pandas.py]: Reference issue number above

**Comment by WillAyd** [comment]
I think you have to catch the warnings now to prevent test failures. You can use #27103 as a reference for how this is done

**Comment by lucaionescu** [comment]
Do you mean using the `@pytest.mark.filterwarnings("ignore:.*msgpack:FutureWarning")`wrapper around the test function this snippet?
 
```python
with catch_warnings():
    filterwarnings('ignore', category=FutureWarning)
```

**Comment by lucaionescu** [comment]
I don't understand why the test fail here, can you point me in the right direction please?

**Comment by gfyoung** [comment]
> I don't understand why the tests fail here

In general, the test results show you every test that fails, so you can go to each one and see why each is failing.  Of course, this is not practical when there are hundreds of failures.

The reason is that there are a lot more tests that pass in `numpy=True` to `read_json` than you have accounted for with this context manager.

@WillAyd : Any thoughts on how we could "apply" this context manager to all of the failing tests?  Copy-pasting that setup across all of them seems a bit tedious 🤷‍♂

**Comment by WillAyd** [comment]
I think if there's a way to limit the filter to the numpy argument at global scope would also work

**Comment by jbrockmendel** [comment]
@lucaionescu can you run `isort -rc pandas`?  i think that will fix the failing CI check

**Comment by jbrockmendel** [comment]
@lucaionescu  can you rebase?  theres a decent chance that the CI failures are unrelated things that have been fixed in the last few days

**Comment by jbrockmendel** [comment]
@lucaionescu can you rebase?

**Comment by lucaionescu** [comment]
Sorry for that! I'm going to fix it ASAP 😄

**Comment by WillAyd** [comment]
@lucaionescu looks like there are no committed changes here - can you take a look?

**Comment by WillAyd** [comment]
@lucaionescu can you take a look here? Would be nice to deprecate this before 1.0

**Comment by jbrockmendel** [comment]
@lucaionescu can you rebase

**Comment by alimcmaster1** [comment]
@lucaionescu let us know if you still want to work on this. If not I can potentially take a look.

Thanks.

**Comment by lucaionescu** [comment]
@alimcmaster1 yes, that would be nice. I don't know what the best way to proceed would be with the failing tests.

**Comment by WillAyd** [comment]
superseded by #30636

**Pull request closed without merging**